src/Misc/baseError.ts

```diff
@@ -1,46 +1,66 @@
 import { ErrorCodesType } from "./errorCodes";
 
 /**
- * BabylonJS error internal class wrapper. Extending Error for a class with docs
- * fails typedoc validation, because lib.es5.d.ts does not have any docs for Error.
+ * See https://stackoverflow.com/questions/12915412/how-do-i-extend-a-host-object-e-g-error-in-typescript
+ * and https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+ *
+ * Polyfill for Object.setPrototypeOf if necessary.
  * @ignore
  */
-class BabylonJSErrorWrapper extends Error {}
+export const setPrototypeOf: (o: any, proto: object | null) => any =
+ (Object as any).setPrototypeOf || ((o, proto) => { o.__proto__ = proto; return o; });
 
 /**
- * Base Error class for babylonjs errors
+ * Base error. Due to limitations of typedoc-check and missing documentation
+ * in lib.es5.d.ts, cannot extend Error directly for RuntimeError.
+ * @ignore
  */
-export abstract class BaseError extends BabylonJSErrorWrapper {
-    // See https://stackoverflow.com/questions/12915412/how-do-i-extend-a-host-object-e-g-error-in-typescript
-    // and https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
-
-    // Polyfill for Object.setPrototypeOf if necessary.
-    protected static _setPrototypeOf: (o: any, proto: object | null) => any =
-        (Object as any).setPrototypeOf || ((o, proto) => { o.__proto__ = proto; return o; });
+abstract class BaseError extends Error {}
 
+/**
+ * Application runtime error
+ */
+ export class RuntimeError extends BaseError {
     /**
      * The error code
      */
-    public errorCode: ErrorCodesType;
+     public errorCode: ErrorCodesType;
 
     /**
-     * The exception that caused the outer exception
+     * Creates a new RuntimeError
+     * @param message defines the message of the error
+     * @param errorCode the error code
      */
-    public innerException?: Error;
+    public constructor(message: string, errorCode: ErrorCodesType) {
+        super(message);
+
+        this.errorCode = errorCode;
+        this.name = "RuntimeError";
+        setPrototypeOf(this, RuntimeError.prototype);
+    }
+}
+
+/**
+ * Application runtime outer error that wraps the cause/innerError
+ */
+export class RuntimeWrapperError extends RuntimeError {

```

Comment:
I know we went back and forth a bit on this. What do you think about putting `innerError` as an optional parameter of `RuntimeError` and then you can remove `RuntimeWrapperError`?